doc/api/v8.md

```diff
@@ -134,3 +134,157 @@ setTimeout(function() { v8.setFlagsFromString('--notrace_gc'); }, 60e3);
 [V8]: https://developers.google.com/v8/
 [here]: https://github.com/thlorenz/v8-flags/blob/master/flags-0.11.md
 [`GetHeapSpaceStatistics`]: https://v8docs.nodesource.com/node-5.0/d5/dda/classv8_1_1_isolate.html#ac673576f24fdc7a33378f8f57e1d13a4
+
+## Serialization API
+
+> Stability: 1 - Experimental
+
+### class: v8.Serializer
+<!--
+added: REPLACEME
+-->
+
+#### new Serializer([options])
+
+* `options` {Object} Set of configurable options to set on the serializer.
+  Can have the following fields:
+  * `getDataCloneError` {Function} A callback that takes a single `message`
+    argument and returns an error object. Defaults to the `Error` constructor.
+  * `writeHostObject` {Function} A callback that takes a single `object`
+    argument and is used to write some kind of host object, if possible. If not,
+    a suitable exception should be thrown.
+
+Creates a new `Serializer` object.
+
+#### serializer.writeHeader()
+
+Writes out a header, which includes the serialization format version.
+
+#### serializer.writeValue(value)
+
+* Returns: {Boolean}
+
+Serializes a JavaScript value and adds the serialized representation to the
+internal buffer.
+
+#### serializer.releaseBuffer()
+
+Returns the stored internal buffer. This serializer should not be used once
+the buffer is released. Calling this method results in undefined behavior
+if a previous write has failed.
```

Comment:
> On failure, you aren't allowed to cleanup?

I guess that depends on what you mean by “cleanup”? Any resources held by the serializer will be returned when it gets gc’ed.

> how do writes fail? do they throw errors? writeHeader/Value don't say they can throw.

Yes, `writeValue` may throw errors. I’ve noted that in the corresponding section.